app/src/main/java/org/oppia/android/app/hintsandsolution/HintsAndSolutionDialogFragmentPresenter.kt

```diff
@@ -121,26 +115,262 @@ class HintsAndSolutionDialogFragmentPresenter @Inject constructor(
+      binding.root.visibility = View.VISIBLE
+      binding.revealHintButton.setOnClickListener {
+        hintsViewModel.isHintRevealed.set(true)
+        expandedHintListIndexListener.onRevealHintClicked(position, /* isHintRevealed= */ true)
+        (fragment.requireActivity() as? RevealHintListener)?.revealHint(
+          saveUserChoice = true,
+          hintIndex = position / 2
+        )
+        val previousIndex: Int? = currentExpandedHintListIndex
+        currentExpandedHintListIndex =
+          if (currentExpandedHintListIndex != null && currentExpandedHintListIndex == position) {
+            null
+          } else {
+            position
+          }
+        expandedHintListIndexListener.onExpandListIconClicked(currentExpandedHintListIndex)
+        if (previousIndex != null && previousIndex != currentExpandedHintListIndex) {
+          bindingAdapter.notifyItemChanged(previousIndex)
+        }
+      }
+    }
+
+    binding.root.setOnClickListener {
+      if (hintsViewModel.isHintRevealed.get()!!) {
+        val previousIndex: Int? = currentExpandedHintListIndex
+        currentExpandedHintListIndex =
+          if (currentExpandedHintListIndex != null && currentExpandedHintListIndex == position) {
+            null
+          } else {
+            position
+          }
+        expandedHintListIndexListener.onExpandListIconClicked(currentExpandedHintListIndex)
+        if (previousIndex != null &&
+          currentExpandedHintListIndex != null &&
+          previousIndex == currentExpandedHintListIndex
+        ) {
+          bindingAdapter.notifyItemChanged(currentExpandedHintListIndex!!)
+        } else {
+          if (previousIndex != null) {
+            bindingAdapter.notifyItemChanged(previousIndex)
+          }
+          if (currentExpandedHintListIndex != null) {
+            bindingAdapter.notifyItemChanged(currentExpandedHintListIndex!!)
+          }
+        }
+      }
+    }
+  }
+
+  private fun bindSolutionViewModel(
+    binding: SolutionSummaryBinding,
+    solutionViewModel: SolutionViewModel
+  ) {
+
+    binding.viewModel = solutionViewModel
+
+    val position: Int = itemList.indexOf(solutionViewModel)
+
+    var isHintListVisible = false
+    if (currentExpandedHintListIndex != null) {
+      isHintListVisible = currentExpandedHintListIndex!! == position
+    }
+    binding.isListExpanded = isHintListVisible
+
+    solutionIndex?.let {
+      isSolutionRevealed?.let {
+        if (solutionIndex == position && isSolutionRevealed!!) {
+          solutionViewModel.isSolutionRevealed.set(true)
+        }
+      }
+    }
+
+    binding.solutionTitle.text = solutionViewModel.title.get()!!.capitalize()
+    // TODO(#1050): Update to display answers for any answer type.
+    if (solutionViewModel.correctAnswer.get().isNullOrEmpty()) {
+      binding.solutionCorrectAnswer.text =
+        """${solutionViewModel.numerator.get()}/${solutionViewModel.denominator.get()}"""
```

Comment:
didn't get this one!!

Reply:
I was refering to this
![image](https://user-images.githubusercontent.com/54740946/119088723-bac61780-ba26-11eb-8f5f-794a8e727bca.png)

Reply:
As there is no static string to concatenate so I think we don't need to shift it to the string res.

Reply:
I don't think this string needs to be a part of `strings.xml` file but on the contrary lets assume that there is some language in which `fractions` is written differently (ideally there won't be any such language) then defining this in strings.xml file is a correct way. 
Now this above example is very unlikely to happen but because right now we cannot confirm this lets define this in `strings.xml` file to be on safe side.

WDYT?

Reply:
sounds good, done